src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -0,0 +1,448 @@
+                $parameters,
+                $requestBody,
+                isset($operation['openapi_context']['callbacks']) ? new \ArrayObject($operation['openapi_context']['callbacks']) : null,
+                $operation['openapi_context']['deprecated'] ?? (bool) $resourceMetadata->getTypedOperationAttribute($operationType, $operationName, 'deprecation_reason', false, true),
+                $operation['openapi_context']['security'] ?? [],
+                $operation['openapi_context']['servers'] ?? []
+            ));
+
+            $paths->addPath($path, $pathItem);
+        }
+
+        return [$links, $schemas];
+    }
+
+    private function buildContent(array $responseMimeTypes, array $operationSchemas): \ArrayObject
+    {
+        $content = new \ArrayObject();
+
+        foreach ($responseMimeTypes as $mimeType => $format) {
+            $content[$mimeType] = new Model\MediaType(new \ArrayObject($operationSchemas[$format]->getArrayCopy(false)));
+        }
+
+        return $content;
+    }
+
+    private function getMimeTypes(string $resourceClass, string $operationName, string $operationType, ResourceMetadata $resourceMetadata = null): array
+    {
+        $requestFormats = $resourceMetadata->getTypedOperationAttribute($operationType, $operationName, 'input_formats', $this->formats, true);
+        $responseFormats = $resourceMetadata->getTypedOperationAttribute($operationType, $operationName, 'output_formats', $this->formats, true);
+
+        $requestMimeTypes = $this->flattenMimeTypes($requestFormats);
+        $responseMimeTypes = $this->flattenMimeTypes($responseFormats);
+
+        return [$requestMimeTypes, $responseMimeTypes];
+    }
+
+    private function flattenMimeTypes(array $responseFormats): array
+    {
+        $responseMimeTypes = [];
+        foreach ($responseFormats as $responseFormat => $mimeTypes) {
+            foreach ($mimeTypes as $mimeType) {
+                $responseMimeTypes[$mimeType] = $responseFormat;
+            }
+        }
+
+        return $responseMimeTypes;
+    }
+
+    /**
+     * Gets the path for an operation.
+     *
+     * If the path ends with the optional _format parameter, it is removed
+     * as optional path parameters are not yet supported.
+     *
+     * @see https://github.com/OAI/OpenAPI-Specification/issues/93
+     */
+    private function getPath(string $resourceShortName, string $operationName, array $operation, string $operationType): string
+    {
+        if ($operation['path'] ?? null) {
+            return 0 === strpos($operation['path'], '/') ? $operation['path'] : '/'.$operation['path'];
+        }
+        $path = $this->operationPathResolver->resolveOperationPath($resourceShortName, $operation, $operationType, $operationName);
+        if ('.{_format}' === substr($path, -10)) {
```

Comment:
Hey @soyuka Does there any way to enable defining '.{_format}'  per resource, even ApiPlatform\Core\OpenApi\Model\Paths does not contain a setter function, in case we decide to decorate the OpenapiFactory  ? I have a resource that should export pdf and docx.
Cheers!